frontend/src/component/Account/UserSpacesConfig.jsx

```diff
@@ -0,0 +1,228 @@
+      { entityType: TLM_ET.SHAREDSPACE_MEMBER, coreEntityType: TLM_CET.MODIFIED, handler: this.handleMemberModified },
+      { entityType: TLM_ET.SHAREDSPACE_MEMBER, coreEntityType: TLM_CET.DELETED, handler: this.handleMemberDeleted }
+    ])
+  }
+
+  handleMemberModified = data => {
+    const { props } = this
+    if (Number(props.userToEditId) !== data.fields.user.user_id) return
+    this.setState(prev => ({
+      workspaceList: prev.workspaceList.map(space =>
+        space.workspace_id === data.fields.workspace.workspace_id
+          ? {
+            ...space,
+            memberList: space.memberList.map(member => member.user_id === Number(props.userToEditId)
+              ? { ...member, do_notify: data.fields.member.do_notify }
+              : member
+            )
+          }
+          : space
+      )
+    }))
+  }
+
+  handleMemberDeleted = data => {
+    const { props } = this
+    if (Number(props.userToEditId) !== data.fields.user.user_id) return
+    this.setState(prev => ({
+      workspaceList: prev.workspaceList.filter(ws =>
+        ws.workspace_id !== data.fields.workspace.workspace_id
+      )
+    }))
+  }
+
+  handleMemberCreated = async data => {
+    const { props } = this
+    if (Number(props.userToEditId) !== data.fields.user.user_id) return
+
+    const newSpace = await this.fillMemberList(data.fields.workspace)
+
+    this.setState(prev => ({
+      workspaceList: [...prev.workspaceList, newSpace].sort(
+        (space1, space2) => space1.workspace_id - space2.workspace_id
+      )
+    }))
+  }
+
+  getWorkspaceList = async () => {
+    const { props } = this
+
+    const fetchGetUserWorkspaceList = await props.dispatch(getUserWorkspaceList(props.userToEditId, false))
+
+    switch (fetchGetUserWorkspaceList.status) {
+      case 200: this.getUserWorkspaceListMemberList(fetchGetUserWorkspaceList.json); break
+      default: props.dispatch(newFlashMessage(props.t('Error while loading user')))
+    }
+  }
+
+  fillMemberList = async space => {
+    const fetchMemberList = await this.props.dispatch(getWorkspaceMemberList(space.workspace_id))
+
+    return {
+      ...space,
+      memberList: fetchMemberList.json || [] // handle error?
+    }
+  }
+
+  getUserWorkspaceListMemberList = async (wsList) => {
+    const workspaceList = await Promise.all(wsList.map(this.fillMemberList))
+    this.setState({ workspaceList })
+  }
+
+  handleConfirmDeleteSpace = async () => {
+    const { props } = this
+    const spaceId = this.state.spaceBeingDeleted
+    if (!spaceId) return
+    this.setState({ spaceBeingDeleted: null })
+
+    const fetchResult = await props.dispatch(deleteWorkspaceMember(spaceId, props.userToEditId))
+    if (fetchResult.status !== 204) {
+      props.dispatch(newFlashMessage(props.t('Error while leaving the space'), 'warning'))
+    }
+  }
+
+  componentDidMount = this.getWorkspaceList
```

Comment:
Does this work?

Reply:
yes :-)

Reply:
It's also a bit confusing to me, it would be easier to call `this.getWorkspaceList` in the `componentDidMount` body

Reply:
ok, done